dato/js/recview.js

```diff
@@ -0,0 +1,158 @@
+    var match = document.cookie.match(matcher);
+    if (match && match[1]) {
+      return match[1].split('.')[0];
+    } else {
+      return false;
+    }
+  }
+
+  function recommenderView() {
+
+    // function to construct element tree for recommender view
+    // if resp.data.response array is empty it skips creating recommender section
+    function renderView(err, resp) {
+      if(err) {
+        console.log(err);
+        return;
+      }
+
+      if(resp.data.response.length){
+        const section = document.createElement('section');
+        section.id = 'recview';
+        const h2 = document.createElement('h2');
+        h2.innerText = 'Recommended Resources';
+        const root = document.createElement('div');
+        root.id = 'root';
+        const list = document.createElement('ul');
+        list.id = 'list';
+
+        root.appendChild(list);
+        section.appendChild(h2);
+        section.appendChild(root);
+        $('#section-').after(section);
+
+        function feedback(event){
+          event.preventDefault();
+          var uuid = resp.data.uuid;
+          var itemId = event.target.href;
+          onClick(uuid, itemId);
+        }
+        for (let i in resp.data.response) {
+          let listEl = document.createElement('li');
+          let link = document.createElement('a');
+          link.className = 'recommendation';
+          link.href = resp.data.response[i].page_url;
+          link.innerText = resp.data.response[i].title;
+          link.addEventListener('click', feedback);
+          listEl.appendChild(link);
+
+          document.getElementById('list').appendChild(listEl);
+        }
+
+        // cb event for mouse over event
+        // tracks when user first hovers over recommender section
+        // sends feedback with data
+        function mouseOverCB(event) {
+          console.log(resp.data.uuid);
+
+          // once we know the schema for what to send back on mouseOver the recommendations
+          // uncomment code and set data
+          var data = {
+            'viewed': 1
+          };
+          client.feedback(resp.data.uuid, { data }, function(err, resp) {
+            if(err) {
+              console.log(err);
+            }
+            console.log(resp);
+          });
+        }
+        $( 'ul#list' ).one( "mouseover", mouseOverCB );
+      }
+
+      // onClick event for links in recommender list
+      // parameters: uuid, and itemId
+      // calls feedback funciton in PS client
+      function onClick(uuid, itemId) {
+        console.log("onClick function");
+        console.log(uuid);
+        console.log(itemId);
+        // Calls client feedback function and sends uuid and feedbackData
+        var data = {'itemId': itemId, 'click': 1};
+        client.feedback(uuid, { data }, function(err, resp) {
+          if(err) {
+            console.log(err);
+          }
+          console.log(resp);
+          window.location.replace(itemId);
```

Comment:
Interesting. In this case I think a comment is warranted (or a variable name change) since the name `itemId` doesn't exactly imply URL usually.

I think we want to use `assign` here rather than `replace`? See http://stackoverflow.com/a/4505846